### PR TITLE
Add private_key_type property in VaultCertificateResponse

### PR DESCRIPTION
**Pull request by abremora**
Fixes gh-676

### CODE REVIEW

**Comment by abremora** [comment]
I added some overloads to be compatible to older versions and marked old methoads as deprecated.

File: spring-vault-core/src/main/java/org/springframework/vault/support/CertificateBundle.java [hunk 1]
```diff
@@ -46,14 +46,43 @@
...
+	 * Create a new {@link CertificateBundle}.
+	 * @param serialNumber the serial number.
+	 * @param certificate the certificate.
+	 * @param issuingCaCertificate the issuing CA certificate.
+	 * @param caChain the CA chain.
+	 * @param privateKey the private key.
+	 * @deprecated since 2.3.3, use {@link #CertificateBundle(String, String, String, List, String, String)} instead.
+	 */
+	@Deprecated
+	CertificateBundle(String serialNumber, String certificate, String issuingCaCertificate,
```
Comment (mp911de) [inline spring-vault-core/src/main/java/org/springframework/vault/support/CertificateBundle.java:63]: We do not need to reintroduce a deprecated constructor variant as the original constructor was package-private so we're free to change the constructors.

File: spring-vault-core/src/main/java/org/springframework/vault/support/CertificateBundle.java [hunk 2]
```diff
@@ -78,13 +110,43 @@ public static CertificateBundle of(String serialNumber, String certificate, Stri
...
 	 * @return the private key (decrypted form, PEM or DER-encoded)
 	 */
 	public String getPrivateKey() {
 		return this.privateKey;
 	}
 
+	/**
+	 * @return the private key type.
+	 */
+	public String getPrivateKeyType() {
```
Comment (mp911de) [inline spring-vault-core/src/main/java/org/springframework/vault/support/CertificateBundle.java:146]: Nit: this needs to be `@Nullable`

**Review by mp911de — APPROVED** [review]
Thanks for submitting a pull request. I'll have a look throughout the next days.

**Comment by abremora** [comment]
@mp911de : Sorry for the delay, I would have continued to work on the PR throughout the day. You already fixed it and pushed it to main branch?

**Comment by mp911de** [comment]
No worries, yeah, that's already on `main`.

**Pull request closed without merging**